packages/browser-pool/src/fingerprinting/hooks.ts

```diff
@@ -63,8 +63,8 @@ export function createPrePageCreateHook() {
         const { fingerprint } = launchContext.fingerprint!;
 
         if (launchContext.useIncognitoPages && browserPlugin instanceof PlaywrightPlugin && pageOptions) {
-            pageOptions.userAgent = fingerprint.navigator.userAgent;
-            pageOptions.viewport = {
+            pageOptions.userAgent ??= fingerprint.navigator.userAgent;
+            pageOptions.viewport ??= {
```

Comment:
With these, the thought process was that we always want a consistent fingerprint (i.e. the use will overwrite the HTTP useragent like this, but won't affect the `navigator.userAgent`, which some bot-detecting services might pick up). 

I'm not too opinionated about this though, if somebody wants to tweak it (and there is an issue to prove this), I guess we should let them.

Reply:
I think we should always respect what users are trying to do, if its a problematic case, we should rather print a warning then silently ignore the options.

Reply:
Agreed - this is just a theoretical case, which shouldn't even cause a hard stop, maybe just an increased rate of blocking + retries. I'm fine with this change, wouldn't even bother with a log message in this case.